### DISCUSSION TITLE
Firebase integration

### DISCUSSION THREAD

**Discussion by illia-romanenko** [GoogleCloudPlatform/functions-framework-dart#274] [Ideas]
Saw a message from @kevmoo in the firebase_interop group about Firebase integration and I think it would be great to start a discussion. I think that if it's available now, we would use it today for:
1. Triggers of all sorts (DB, auth/user)
2. Cron
3. Callable/HTTP functions

**Comment by jimmyff** [comment]
Hi @illia-romanenko, 
I think the majority of what you're wanting is already available:

>     1. Triggers of all sorts (DB, auth/user)
Event Arc + Cloud run seems to be the solution for this, although for firestore events it's bleedin-edge. See this discussion for details: https://github.com/GoogleCloudPlatform/functions-framework-dart/discussions/249
>     2. Cron
Scheduler would be the solution for this: https://cloud.google.com/scheduler/docs/creating
>     3. Callable/HTTP functions
I'm using Cloud Run http instances for this. My flutter app calls these rather than firestore cloud functions. The only complication is I need to authorise the Firebase authentication JWT token myself. GCP docs relating to this are here:
https://firebase.google.com/docs/auth/admin/verify-id-tokens#verify_id_tokens_using_a_third-party_jwt_library
I used the following packages:  `jose` & `x509`. Jose does most of the work and has a `JsonWebToken` class which provides a `verify` method.

**Reply by ericq** [reply]
can you share where do you place the code to initialize the firebase store connection. It should be outside of the function, but where?
